modules/dr-restoring-cluster-state.adoc

```diff
@@ -304,3 +304,21 @@ etcd-ip-10-0-173-171.ec2.internal                2/2     Running     0
 ----
 
 Note that it might take several minutes after completing this procedure for all services to be restored. For example, authentication by using `oc login` might not immediately work until the OAuth server pods are restarted.
+
+[id="dr-scenario-2-restoring-cluster-state_storage_{context}"]
+= Issues and workarounds in restoring persistent storage state
+
+If your {product-title} uses persistent storage of any form, typically a state of the world is stored outside etcd. It could be a ElasticSearch
+cluster running in a Pod or a database running in a statefulset. When {product-title} is restored from etcd backup, we are also restoring status of those workloads in Openshift. But depending on how old the etcd snapshot is - that status may be invalid or outdated.
+
+Some examples are:
+
+1. Mysql database was running in a Pod backed up by a PV object. But after the etcd snapshot was taken - the Mysql Pod and corresponding PV were deleted and hence underlying volume on storage provider would be deleted. Restoring Openshift from etcd snapshot is not going to bring back the volume on storage provider and hence is not going to result in running Mysql Pod(even though Pod will try to start but will keep failing) and a manual action must be taken to restore mysql Pod. In this case manual action could be restoring the volume on storage provider via volume snapshots and then editing PV object to point to new volume.
+
+2. You have a pod P1 running using volume A which is attached to node X. If the etcd snapshot is taken and another pod uses the same volume on node Y and a etcd restore is performed then pod P1 may not able to start correctly because volume is still attached to node Y but {product-title} does know about it and hence is not going to detach it. In this case volume must be manually detached from node Y, so as volume can attach on node X and pod P1 can start.
+
+3. If cloudprovider or storage provider credentials were updated since etcd snapshot was taken, any CSI drivers or operators that depend on the those credentials will not work and hence you may have to manually update credentials required by those drivers or operators.
+
+4. Local Storage Operator creates symlinks for each persistent volume it manages from `/dev/disk/by-id` or `/dev` directories. If a device is removed or renamed from {product-title} nodes since etcd snapshot was taken, the local volume Persistent Volumes(PVs) may refer to devices which no longer exist. To workaround this problem - admin must manually remove the PVs with invalid devices and remove symlink from respective nodes and delete `LocalVolume` or `LocalVolumeSet` objects. See section on deleting Local Storage Operator Resources in {product-title} documentation.
+
+5. Contents of Persistent Volumes(PVs) is never part of etcd snapshots and hence when restoring {product-title} cluster from etcd snapshot could result in non-critical workloads to gain access to critical data (or vice-versa).
```

Comment:
We should be defining PV above this point, so I think it's safe to remove the definition here.

I'd also suggest breaking this into multiple sentences - "The contents of PVs are never part of the etcd snapshot. When restoring an {product-title} cluster from the etcd snapshot it could result in non-critical workloads gaining access to critical data (or vice-versa)."

Reply:
I'd change to: "The contents of PVs are never part of the etcd snapshot. When restoring an {product-title} cluster from the etcd snapshot, non-critical workloads could gain access to critical data (or vice-versa)."

However, point 5 seems to be different from the preceding 4 points. It's more of warning than about a scenario that creates a stale/wrong etcd. I'd pull this out of the list and make it an "Important" admonition.